facebook-commerce.php

```diff
@@ -997,8 +1037,9 @@ private function save_product_settings( \WC_Product $product ) {
 	 * Deletes a product from Facebook.
 	 *
 	 * @param int $product_id product ID
+	 * @param bool $force_delete boolean whether to foce delete product from Facebook or not
 	 */
-	public function on_product_delete( $product_id ) {
+	public function on_product_delete( $product_id, $force_delete = false ) {
```

Comment:
I think the purpose of the `$force_delete` boolean parameter is to bypass the verifications at the beginning of the function. If that's the case, can we remove the boolean parameter and extract [these lines](https://github.com/facebookincubator/facebook-for-woocommerce/blob/2325786dad17b5fa6b5cdc0c766ebb4a744d034c/facebook-commerce.php#L1062-L1091) into a private helper that we can call directly instead?

Reply:
Also, I think we need to remove the stored product item ID and product group ID when we delete a product from the catalog to avoid errors the next time you try to sync the deleted product: https://cloud.skyver.ge/L1uQZAgx

I got that one on a simple product:

1. Set Facebook sync to Do not sync and click the button to remove from sync and delete
1. Update the product
1. Now set Facebook sync to Sync and show on catalog
1. Update the product
1. The error notice should show up when the page loads

Reply:
@wvega 

> I think the purpose of the $force_delete boolean parameter is to bypass the verifications at the beginning of the function. If that's the case, can we remove the boolean parameter and extract these lines into a private helper that we can call directly instead?

Make sense, I addressed that in these commits 75612c3 & fa3c984